libs/animation/animation.ts

```diff
@@ -1,173 +1,624 @@
+                    if (metadata.lastNode instanceof QuadraticCurveTo) {
+                        lastControlPoint = (metadata.lastNode as QuadraticCurveTo).p1;
+                    } else if (metadata.lastNode instanceof CubicCurveTo) {
+                        lastControlPoint = (metadata.lastNode as CubicCurveTo).p2;
+                    } else break;
+
+                    const p1 = new Point(p0.x + (p0.x - lastControlPoint.x), p0.y + (p0.y - lastControlPoint.y));
+                    const p2 = new Point(args[0], args[1]);
+                    const p3 = new Point(args[2], args[3]);
+                    node = new CubicCurveTo(p0, p1, p2, p3);
+                    break;
+                }
+                case "s": { // s dx2 dy2 dx3 dy3
+                    let lastControlPoint: Point;
+                    if (metadata.lastNode instanceof QuadraticCurveTo) {
+                        lastControlPoint = (metadata.lastNode as QuadraticCurveTo).p1;
+                    } else if (metadata.lastNode instanceof CubicCurveTo) {
+                        lastControlPoint = (metadata.lastNode as CubicCurveTo).p2;
+                    } else break;
+
+                    const p1 = new Point(p0.x + (p0.x - lastControlPoint.x), p0.y + (p0.y - lastControlPoint.y));
+                    const p2 = new Point(p0.x + args[0], p0.y + args[1]);
+                    const p3 = new Point(p0.x + args[2], p0.y + args[3]);
+                    node = new CubicCurveTo(p0, p1, p2, p3);
+                    break;
+                }
+                case "Z": // Z
+                case "z": { // z
+                    node = new LineTo(p0, metadata.pathStart);
+                    break;
+                }
             }
-            animations.push(this);
+
+            return node;
         }
 
-        update() {
-            let currentTime = control.millis();
-            let dt = currentTime - this.lastTime;
-            if (dt >= this.interval && this.frames.length) {
-                this.index = (this.index + 1) % this.frames.length;
-                this.lastTime = currentTime;
-            }
-            
-            this.sprites = this.sprites.filter(sprite => !(sprite.flags & sprites.Flag.Destroyed));
-
-            this.sprites.forEach(sprite => {
-                if (sprite._action === this.action) {
-                    let newImage = this.getImage();
-                    //Update only if the image has changed
-                    if (sprite.image !== newImage) {
-                        sprite.setImage(newImage);
+        public static parse(pathStart: Point, pathString: string): Path {
+            let path: Path = new Path();
+            let p0: Point = pathStart;
+
+            // This implementation of SVG parsing does not support the A/a commands, nor does it support exponents in arguments
+            const digits = "0123456789";
+            const separators = ", \t\n\r\f\v";
+            const commands: {
```

Comment:
Might be worth pulling this out into a helper function that just returns based off a switch statement, obj literals are much slower and take up much more space on hw

Reply:
Could you clarify which parts would be pulled into their own helper function? What exactly would the function be accomplishing? Or would it be multiple, like isDigit, isSeparator, etc.?

Reply:
something like 

```typescript
function commandToArgCount(command: string): number {
    switch (command) {
        case "M":
        case "m":
            return 4;
        ...
        default: return -1
    }
}
```

just to replace commands here